contracts/oracles/common/CorrelatedTokenOracle.sol

```diff
@@ -19,13 +22,33 @@ abstract contract CorrelatedTokenOracle is CappedOracle {
     /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
     address public immutable UNDERLYING_TOKEN;
 
+    //// @notice Growth rate percentage in seconds. Ex: 1e18 is 100%
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable GROWTH_RATE_PER_SECOND;
+
+    /// @notice Snapshot update interval
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SNAPSHOT_INTERVAL;
+
     /// @notice Address of Resilient Oracle
     /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
     OracleInterface public immutable RESILIENT_ORACLE;
 
+    /// @notice Last stored snapshot price
+    uint256 public snapshotPrice;
```

Comment:
I think we should rename this to `snapshotExchangeRate` to be clear about the stored value